akka-http-core/src/main/scala/akka/http/impl/util/EnhancedInetSocketAddress.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.impl.util
+
+import java.lang.reflect.{ InvocationTargetException, Method }
+import java.net.InetSocketAddress
+
+import scala.util.control.NonFatal
+
+/**
+ * Provides getHostString support for Java 6.
+ *
+ * TODO: can be removed once support for Java 6 is dropped.
+ *
+ * Internal API
+ */
+private[http] class EnhancedInetSocketAddress(val address: InetSocketAddress) extends AnyVal {
```

Comment:
Added a ticket to migrate this into 2.3: https://github.com/akka/akka/issues/17701